[docs-only] add release roadmap (#1393)
owncloud/ocis: docs/ocis/release_roadmap.md

We are going to stick to major version 1 until we feel confident about running oCIS to production environments. As a consequence of this we cannot raise the major version, like SemVers requires it, even if we need to introduce breaking changes. We will do our best to avoid breaking changes. If there is no way to circumvent this, we will add an automatic migration or at least point out manual migration steps, since we as oCIS developers are already using oCIS on a personal basis. The best place to see if a breaking change happens is our changelog which is available for every release. If things are not working out for you please contact us immediately. We want to know about this and solve it for you.

It isn't our intention to scare you with our addendum "Tech Preview". We want you to have a clear picture of what you can expect from oCIS. You could take it as a disclaimer or even compare it like running an Linux kernel in alpha stage. It can be very pleasing to be on the latest codebase but you could also find yourself with a lot of problems arising because of that.

> You clearly can expect a totally new experience of file-sync and share with oCIS and want you to use it now - but with understanding and caution.

Reviewer: ```suggestion
You clearly can expect a totally new experience of file-sync and share with oCIS and we want you to use it now - but with understanding and caution.
```

Reviewer: Or maybe:
```suggestion
You clearly can expect a totally new experience of file-sync and share with oCIS and we want you to try it now - but with understanding and caution.
```
What level between "try" and "use" to we want to suggest is "reasonable"?

Author: I think at some point we want people more than trying it out once. We want them to use it. But like in my example with the alpha stage kernel they should feel confident about what they are doing and understanding the involved risks.

## ownCloud Infinite Scale 2.x general availability releases

> Starting with oCIS 2.0.0 release we will strictly stick to SemVer, just as we do right now for ownCloud Server. The general availability release also means that we can recommend oCIS warmly to everyone. Use it to store your precious family pictures or you confidential company data!

Reviewer: ```suggestion
Starting with oCIS 2.0.0 release we will strictly stick to SemVer, just as we do right now for ownCloud Server. The general availability release will also mean that we can recommend oCIS warmly to everyone. Use it to store your precious family pictures or you confidential company data!
```
